### DISCUSSION TITLE
Paginator

### DISCUSSION THREAD

**Discussion by Tartali** [danirus/django-comments-xtd#369] [Q&A]
How can I use the paginator for comments? Is it even possible?

**Comment by danirus** [comment]
The paginator for comments is only available in v3.0.0, which is still a work-in-progress. The paginator is done, but the complete version is not finished yet, and the documentation is not ready. Nevertheless you can see how it works in the `demos/project-stories`, in the branch v3.0, there is a [README.md](https://github.com/danirus/django-comments-xtd/tree/rel-3.0.0/demos/project-stories) with instructions on how to set up that demo project.

About the paginator, you can take a look at the [paginator.py](https://github.com/danirus/django-comments-xtd/blob/rel-3.0.0/django_comments_xtd/paginator.py) module, it has a long doc string explaining how it works, it might be a bit obscure, because it deals with comments that can have nested comments. Basically, the paginator never splits a comment thread. That's why it has the behaviour explained in the doc string.

The reason why v2 does not have a paginator is that v2 uses a tree to render comments, while v3 uses a list. In v3 we use `render_xtdcomment_list` instead of v2's `render_xtdcomment_tree`.

Let me know if you encounter any problem with the `demo/project-stories` or if you have questions about the paginator. I hope to be able to release v3 sometime in the near future... 🙂

**Reply by Tartali** [reply]
Well, I will wait for V3.0. I hope to wait 🙂
